packages/yourloops/pages/hcp/team-edit-modal.tsx

```diff
@@ -0,0 +1,240 @@
+      backgroundColor: theme.palette.background.paper,
+      borderRadius: theme.shape.borderRadius,
+      width: theme.breakpoints.width("sm"),
+    },
+    form: {
+      display: "flex",
+      flexDirection: "column",
+      padding: theme.spacing(2),
+      maxHeight: "30em",
+      overflowY: "scroll",
+    },
+    formChild: {
+      marginBottom: theme.spacing(2),
+    },
+    divModalButtons: {
+      display: "flex",
+      flexDirection: "row",
+      marginTop: theme.spacing(2),
+      marginRight: theme.spacing(4), // eslint-disable-line no-magic-numbers
+    },
+    divModalButtonCancel: {
+      marginLeft: "auto",
+      marginRight: theme.spacing(1),
+    },
+  };
+});
+
+function TeamEditModal(props: TeamEditModalProps): JSX.Element {
+  const { action, team, modalOpened, setModalOpen, onSaveTeam } = props;
+  const modalBackdropTimeout = 300;
+
+  const classes = modalStyles();
+  const auth = useAuth();
+
+  console.info("TeamEditModal", auth.user?.settings?.country);
+
+  const [ teamName, setTeamName ] = React.useState(team.name ?? "");
+  const [ teamPhone, setTeamPhone ] = React.useState(team.phone ?? "");
+  const [ teamEmail, setTeamEmail ] = React.useState(team.email ?? "");
+  const [ addrLine1, setAddrLine1 ] = React.useState(team.address?.line1 ?? "");
+  const [ addrLine2, setAddrLine2 ] = React.useState(team.address?.line2 ?? "");
+  const [ addrZipCode, setAddrZipCode ] = React.useState(team.address?.zip ?? "");
+  const [ addrCity, setAddrCity ] = React.useState(team.address?.city ?? "");
+  const [ addrCountry, setAddrCountry ] = React.useState(auth.user?.settings?.country ?? "FR");
+  const [ formIsIncomplete, setFormIsIncomplete ] = React.useState(true);
+
+  const ariaModal = action === "create" ? t("aria-modal-team-add") : t("aria-modal-team-edit");
+  const modalTitle = action === "create" ? t("modal-team-add-title") : t("modal-team-edit-title");
+  const modalButtonValidate = action === "create" ? t("modal-team-button-create") : t("modal-team-button-edit");
+
+  const countries: LocalesCountries = locales.countries;
+  const optionsCountries: JSX.Element[] = [];
+  for (const entry in countries) {
+    if (Object.prototype.hasOwnProperty.call(countries, entry)) {
+      const { name } = countries[entry];
+      optionsCountries.push(<option value={entry} key={name} aria-label={name}>{name}</option>);
+    }
+  }
+  optionsCountries.sort((a: JSX.Element, b: JSX.Element) => {
+    const aName = a.key as string;
+    const bName = b.key as string;
+    return aName.localeCompare(bName);
+  });
+
+  const isFormIsIncomplete = (): boolean => {
+    let valid = teamName.length > 1;
+    valid = valid && teamPhone.length > 1;
+    valid = valid && addrLine1.length > 1;
+    valid = valid && addrZipCode.length > 1;
+    valid = valid && addrCity.length > 1;
+    valid = valid && addrCountry.length > 1;
+    valid = valid && (teamEmail.length < 1 || REGEX_EMAIL.test(teamEmail));
+    return !valid;
+  };
```

Comment:
I would at least manage  a max length for the team Name.
But I guess it's something we may have to adjust in next versions with specific conditions and warning messages. But we can keep it as it is.

Reply:
Good idea.
Same for all the fields